python/lsst/summit/utils/guiders/plotting.py

```diff
@@ -351,81 +431,98 @@ def _starMosaic(
-                center=refCenter,
+                viewCenter=starInfo.refCenter,
                 cutoutSize=cutoutSize,
                 plo=plo,
                 phi=phi,
-                annotate=False,
             )
-            # Static overlays (when not animating)
-            if not isAnimated:
-                addStaticOverlays(axs[detName], detName, centerCutout, cutoutSize, camRotAngle=camAngle)
-
             artists.append(imObj)
 
-            if withStars:
-                starCross = plotStarCentroid(
-                    axs[detName],
-                    centerCutout,
-                    deltaXY=centroidOffset,
-                    markerSize=8 if cutoutSize > 0 else 0,
+            # Static overlays (reference frame elements)
+            if not isAnimated:
+                addReferenceOverlays(ax, detName, starInfo.refCenter, cutoutSize)
+
+            # Animated overlays (star position elements)
+            if starInfo.hasData and cutoutSize > 0:
+                # Rotated crosshair follows the star
+                crosshairArtists = plotCrosshairRotated(
+                    starInfo.starCenter,
+                    90 + camAngle,
+                    ax,
+                    color="grey",
+                    size=cutoutSize,
                 )
-                jitter = getStdCentroid(self.starsDf, self.expId)
+                artists.extend(crosshairArtists)
 
+                # Star centroid marker
+                starCross = plotStarCentroid(
+                    ax,
+                    starInfo.starCenter,
+                    markerSize=8,
+                )
                 artists.extend(starCross)
 
-            cutoutShapeList.append(shape)
+                jitter = getStdCentroid(self.starsDf, self.expId)
 
+        # Center panel annotation
         stampInfo = annotateStampInfo(
             axs["center"], expid=self.expId, stampNum=stampNum, nStamps=nStamps, view=view, jitter=jitter
         )
         artists.append(stampInfo)
 
-        cutoutSize = np.max(cutoutShapeList) if cutoutShapeList else 30
-
+        # Arrow panel
         if not isAnimated:
-            drawArrows(axs["arrow"], cutoutSize, 90.0 + self.camRotAngle)
+            drawArrows(axs["arrow"], cutoutSize if cutoutSize > 0 else 30, 90.0 + self.camRotAngle)
 
-        for ax in axs.values():
-            clearAxisTicks(ax, isSpine=cutoutSize < 0)
+        # Clear ticks - detector panels keep spines for full frame,
+        # center/arrow never have spines
+        for name, ax in axs.items():
+            isDetector = name not in ("center", "arrow")
+            clearAxisTicks(ax, isSpine=isDetector and cutoutSize < 0)
 
         if saveAs:
             fig.savefig(saveAs, dpi=120)
 
         return artists
 
+    def _getStarInfo(self, detName: str, stampNum: int) -> StarInfo:
+        """Get star info for a detector, falling back to image center
+        if unavailable.
+        """
+        if not self.withStars:
```

Comment:
Is it expected that `_getStarInfo` should be called with `self.withStars` is `False`? If not, then I'd at least log a warning. More than that though, could you explain a bit about why falling back to the image center is functionality you want here? Naively, I'd have thought that if you want into on a star's position, and that's not possible, you shouldn't just use the chip center.